modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/RetrieveAppPropertyTestCase.java

```diff
@@ -0,0 +1,95 @@
+
+public class RetrieveAppPropertyTestCase {
+    private static final String TEST_DESCRIPTION = "Verify Retrieving Web Apps Properties";
+    private APPMPublisherRestClient appmPublisherRestClient;
+    private String appName = "RetrieveAppPropertyTestCase";
+    private String appVersion = "1.0.0";
+    private String context = "/" + appName;
+    private String trackingCode = "AM_" + appName;
+    private User adminUser;
+    private String userName;
+    private String password;
+    private String backEndUrl;
+
+
+    @BeforeClass(alwaysRun = true)
+    public void startUp() throws Exception {
+        AutomationContext appMServer = new AutomationContext(AppmTestConstants.APP_MANAGER,
+                                                             TestUserMode.SUPER_TENANT_ADMIN);
+        backEndUrl = appMServer.getContextUrls().getWebAppURLHttps();
+        appmPublisherRestClient = new APPMPublisherRestClient(backEndUrl);
+
+        adminUser = appMServer.getSuperTenant().getTenantAdmin();
+        userName = adminUser.getUserName();
+        password = adminUser.getPassword();
+
+        appmPublisherRestClient.login(userName, password);
+    }
+
+    @Test(description = TEST_DESCRIPTION)
+    public void testAppPropertyRetrieval() throws Exception {
+        HttpResponse response = appmPublisherRestClient.webAppCreate(appName, context, appVersion,
+                                                                     trackingCode);
+        JSONObject responseData = new JSONObject(response.getData());
+        String uuid = responseData.getString(AppmTestConstants.ID);
+        String appType = AppmTestConstants.WEB_APP;
+        appmPublisherRestClient.publishWebApp(uuid);
+
+        HttpResponse appPropertyResponse = appmPublisherRestClient.getWebAppProperty(uuid);
+        JSONObject jsonObject = new JSONObject(appPropertyResponse.getData());
+
+        //Check App Id
+        String appId = (String) jsonObject.get(AppmTestConstants.ID);
+        assertTrue((appId.equals(uuid) == true), "Unable to Retrieve application id.");
+
+        //Check App Type
+        String type = (String) jsonObject.get(AppmTestConstants.TYPE);
+        assertTrue((type.equals(appType) == true), "Unable to Retrieve application type.");
+
+        //Check lifecycleState
+        String lifecycleState = (String) jsonObject.get(AppmTestConstants.LIFE_CYCLE_STATE);
+        assertTrue((lifecycleState.equalsIgnoreCase(AppmTestConstants.PUBLISHED) == true),
+                   "Unable to Retrieve application life cycle.");
+
+        //Check Path attribute
+        String appPropertyString = userName + "/" + appName + "/" + appVersion + "/" + appType;
+        String path = (String) jsonObject.get("path");
+        assertTrue(path.endsWith(appPropertyString), "Unable to Retrieve application path.");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void closeDown() throws Exception {
+        appmPublisherRestClient.logout();
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/commit/f021f86c3c398dddeeca17e225a37e42f7f7033b